### DISCUSSION TITLE
how to add customed score function to tune_model

### DISCUSSION THREAD

**Discussion by awakumaya** [pycaret/pycaret#3191] [Q&A]
Hi, all

I'm stacked at adding customed score funcion to tune_mode() as an arg of "custom_scorer". My scripts with error is 

```
 def calc_auc(y_true, y_pred):
     auc = roc_auc_score(y_true, y_pred, average='macro', multi_class = 'ovr')
     return auc
     
 auc_scorer = make_scorer(calc_auc)
 
 tuned_model = tune_model(estimator=model, fold=5, custom_grid = params, custom_scorer='auc_scorer')
```
Emerged error message was the below:
> ValueError: Optimize method not supported. See docstring for list of available parameters.

I saw an article that an answer solve a similar trouble to mine. So I tried again in accoding to the solution (https://github.com/pycaret/pycaret/issues/876) like this:

```
add_metric(id='auc_ovr', name='AUC OvR', score_func=calc_auc, greater_is_better=True, multiclass=True) 
tuned_model = tune_model(estimator=model, fold=5, custom_grid = params, custom_scorer='auc_ovr')
```

The above script finish with no error. But all values calculated from the customed funciton were '0.0000', although the other metrics are output correctly.

Could anyone help me?

**Comment by jshahbazi** [comment]
I believe this is a bug in the new version.  It doesn't work at all for me, even when using something like this:
```
from sklearn.metrics import log_loss
add_metric('logloss', 'Log Loss', log_loss, greater_is_better = False)
```

**Reply by ngupta23** [reply]
Which version are you referring to?

**Reply by awakumaya** [reply]
I'm using pycaret with 2.3.10 and scikit-learn with 0.23.2 on jupyterlab dektop windows. And I also tried logloss function by uing add_metric, like jshahbazi told. But the all results of the logloss were zero.

**Reply by jshahbazi** [reply]
Version 3.0-rc6.  I'm on macOS.

**Reply by jshahbazi** [reply]
@ngupta23 I've opened an issue for this: https://github.com/pycaret/pycaret/issues/3210
